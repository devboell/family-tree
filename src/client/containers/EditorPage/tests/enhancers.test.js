import { print } from 'graphql'

import {
  PERSONS_QUERY,
  RELATIONS_QUERY,
  CREATE_PERSON_MUTATION,
  UPDATE_PERSON_MUTATION,
  REMOVE_PERSON_MUTATION,
} from '../enhancers'

it('PERSONS_QUERY, has a known query shape', () => {
  expect(print(PERSONS_QUERY)).toMatchSnapshot()
})

it('RELATIONS_QUERY, has a known query shape', () => {
  expect(print(RELATIONS_QUERY)).toMatchSnapshot()
})

it('CREATE_PERSON_MUTATION, has a known query shape', () => {
  expect(print(CREATE_PERSON_MUTATION)).toMatchSnapshot()
})

it('UPDATE_PERSON_MUTATION, has a known query shape', () => {
  expect(print(UPDATE_PERSON_MUTATION)).toMatchSnapshot()
})

it('REMOVE_PERSON_MUTATION, has a known query shape', () => {
  expect(print(REMOVE_PERSON_MUTATION)).toMatchSnapshot()
})
