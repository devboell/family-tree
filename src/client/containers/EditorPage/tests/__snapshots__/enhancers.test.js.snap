// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CREATE_PERSON_MUTATION, has a known query shape 1`] = `
"mutation createPersonMutation($name: String, $title: String, $birthDate: Int, $deathDate: Int, $gender: String, $isMonarch: Boolean, $bornToId: ID, $partners: [PartnerInput]) {
  createPerson(name: $name, title: $title, birthDate: $birthDate, deathDate: $deathDate, gender: $gender, isMonarch: $isMonarch, bornToId: $bornToId, partners: $partners) {
    id
    ...personInfo
  }
}

fragment personInfo on Person {
  name
  title
  birthDate
  deathDate
  gender
  isMonarch
  bornToId
  partnerIds
  partners {
    relationId
    partner {
      id
      name
    }
    children {
      name
    }
  }
}
"
`;

exports[`PERSONS_QUERY, has a known query shape 1`] = `
"query personsQuery {
  persons {
    id
    ...personInfo
  }
}

fragment personInfo on Person {
  name
  title
  birthDate
  deathDate
  gender
  isMonarch
  bornToId
  partnerIds
  partners {
    relationId
    partner {
      id
      name
    }
    children {
      name
    }
  }
}
"
`;

exports[`RELATIONS_QUERY, has a known query shape 1`] = `
"query relationsQuery {
  relations {
    id
    person1 {
      name
    }
    person2 {
      name
    }
  }
}
"
`;

exports[`REMOVE_PERSON_MUTATION, has a known query shape 1`] = `undefined`;

exports[`UPDATE_PERSON_MUTATION, has a known query shape 1`] = `
"mutation updatePersonMutation($id: ID!, $name: String, $title: String, $birthDate: Int, $deathDate: Int, $gender: String, $isMonarch: Boolean, $bornToId: ID, $partners: [PartnerInput]) {
  updatePerson(id: $id, name: $name, title: $title, birthDate: $birthDate, deathDate: $deathDate, gender: $gender, isMonarch: $isMonarch, bornToId: $bornToId, partners: $partners) {
    id
    ...personInfo
  }
}

fragment personInfo on Person {
  name
  title
  birthDate
  deathDate
  gender
  isMonarch
  bornToId
  partnerIds
  partners {
    relationId
    partner {
      id
      name
    }
    children {
      name
    }
  }
}
"
`;
